test/addBinding.test.ts

```diff
@@ -0,0 +1,78 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { commands, Uri } from 'vscode';
+import { AzExtTreeItem } from 'vscode-azureextensionui';
+import { ext, getRandomHexString, IFunctionBinding, IFunctionJson, ProjectLanguage } from '../extension.bundle';
+import { createTestActionContext, testUserInput, testWorkspacePath } from './global.test';
+
+suite('Add Binding', async () => {
+    let functionJsonPath: string;
+    const functionName: string = 'HttpTriggerTest';
+    let initialBindingsCount: number;
+
+    suiteSetup(async () => {
+        await fse.emptyDir(testWorkspacePath);
+        await testUserInput.runWithInputs([testWorkspacePath, ProjectLanguage.JavaScript, /http\s*trigger/i, functionName, 'Anonymous'], async () => {
+            await commands.executeCommand('azureFunctions.createNewProject');
+        });
+        functionJsonPath = path.join(testWorkspacePath, functionName, 'function.json');
+        assert.ok(await fse.pathExists(functionJsonPath), 'Failed to create project');
+        initialBindingsCount = await getBindingsCount();
+    });
+
+    suiteTeardown(async () => {
+        const finalBindingsCount: number = await getBindingsCount();
+        assert.equal(finalBindingsCount, initialBindingsCount + 3, 'Not all expected bindings were added.');
+    });
+
+    test('Command Palette', async () => {
+        const userInputs: string[] = [functionName];
+        // https://github.com/microsoft/vscode-azurefunctions/issues/1586
+        const isLoggedOut: boolean = (await ext.azureAccountTreeItem.getCachedChildren(createTestActionContext())).some(s => /sign in/i.test(s.label));
+        if (isLoggedOut) {
+            userInputs.unshift('Local Project');
+        }
+        await validateAddBinding([], userInputs);
+    });
+
+    test('Uri', async () => {
+        await validateAddBinding([Uri.parse(functionJsonPath)], []);
+    });
+
+    test('Tree', async () => {
+        const treeItem: AzExtTreeItem | undefined = await ext.tree.findTreeItem('/localProjecttestWorkspace/functions/HttpTriggerTest/bindings', createTestActionContext());
```

Comment:
nit: I don't really think we'd likely change functionName from "HttpTriggerTest", but you could use the variable here instead of hardcoding it